Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -0,0 +1,394 @@
+    end
+
+    # Cache demodulized strategy names, to avoid repeating this work
+    Strategy.constants.sort.each do |strategy_symbol|
+      strategy = Strategy.const_get(strategy_symbol)
+      @livecheck_strategy_names[strategy] = strategy.name.demodulize
+    end
+    @livecheck_strategy_names.freeze
+
+    has_a_newer_upstream_version = false
+    formulae_checked = formulae_to_check.sort.map.with_index do |formula, i|
+      puts "\n----------\n" if args.debug? && i.positive?
+
+      skip_result = skip_conditions(formula, args: args)
+      next skip_result if skip_result != false
+
+      formula.head.downloader.shutup! unless formula.stable?
+
+      current = formula.stable? ? formula.version : formula.installed_version.version.commit
+
+      version_info = nil
+      latest = if formula.stable?
+        version_info = latest_version(formula, args: args)
+        version_info.nil? ? nil : version_info["latest"]
+      else
+        formula.head.downloader.fetch_last_commit
+      end
+
+      if latest.nil?
+        raise TypeError, NO_VERSIONS_MSG unless args.json?
+
+        next version_info if version_info.is_a?(Hash) && version_info["status"] && version_info["messages"]
+
+        next status_hash(formula, "error", [NO_VERSIONS_MSG], args: args)
+      end
+
+      if (m = latest.to_s.match(/(.*)-release$/)) && !current.to_s.match(/.*-release$/)
+        latest = Version.new(m[1])
+      end
+
+      info = {
+        "formula" => formula_name(formula, args: args),
+        "version" => {
+          "current"             => current.to_s,
+          "latest"              => latest.to_s,
+          "outdated"            => formula.stable? ? (current < latest) : (current != latest),
+          "newer_than_upstream" => formula.stable? && (current > latest),
+        },
+        "meta"    => {
+          "livecheckable" => formula.livecheckable?,
+        },
+      }
+      info["meta"]["head_only"] = !formula.stable? unless formula.stable?
+      info["meta"].merge!(version_info["meta"]) unless version_info.nil? || !version_info.key?("meta")
+
+      next if args.newer_only? && !info["version"]["outdated"]
+
+      has_a_newer_upstream_version ||= true
+
+      if args.json?
+        info.except!("meta") unless args.verbose?
+        next info
+      end
+
+      print_latest_version(info, args: args)
+      nil
+    rescue => e
+      Homebrew.failed = true
+
+      if args.json?
+        status_hash(formula, "error", [e.to_s], args: args)
+      elsif !args.quiet?
+        onoe "#{Tty.blue}#{formula_name(formula, args: args)}#{Tty.reset}: #{e}"
+        nil
+      end
+    end
+
+    if (args.newer_only? && !has_a_newer_upstream_version) && !(args.json? || args.debug?)
+      puts "No newer upstream versions."
+    end
+
+    puts JSON.generate(formulae_checked.compact) if args.json?
+  end
+
+  def formula_name(formula, args:)
+    args.full_name? ? formula.full_name : formula.name
+  end
+  private_class_method :formula_name
```

Comment:
Would anywhere mean methods in `livecheck/strategy.rb` or individual strategies as well? Just confirming because we have a `private_class_method :strategies` in `unpack_strategy.rb`.